io/net/bonding.py

```diff
@@ -339,12 +339,10 @@ def bond_fail(self, arg1):
                                                          self.remotehost)
                 if peer_networkinterface.set_mtu(mtu) is not None:
                     self.cancel("Failed to set mtu in peer")
-            if self.ping_check():
-                self.log.info("Ping passed for Mode %s", self.mode,
-                              mtu)
+            if not self.ping_check():
+                self.fail("Ping fail in mode %s after MTU change to %s" % (self.mode, mtu))
             else:
-                error_str = "Ping fail in Mode %s after MTU change to %s"\
-                    % (arg1, mtu)
+                self.log.info("Ping success for mode %s bond with  MTU %s" % (self.mode, mtu))
             if self.bond_networkinterface.set_mtu('1500'):
```

Comment:
Its the same way I see 
old
if pingworks:
    pass message
else:
    fail message

new changes

if not ping pass:
   fail message
else
   pass message

Reply:
ah you have added log.fail

Reply:
Test was not reporting FAIL when there was packet loss /ping failed. It always showed PASS in case of PASS or FAIL
added `self.fail("...." )` after reordering the if..else condition